File: ConstructionWorkers/src/ASTnodes/sections/Question.java
package ASTnodes.sections;

/**
 * Created by LGGX on 10-Feb-17.
 */
import ASTnodes.CodeLocation;
import ASTnodes.expressions.literals.IDENTIFIER;
import ASTnodes.visitors.AllVisitors;

import ASTnodes.types.Type;

public class Question extends Section {

    private final IDENTIFIER identifier;
    private final String label;
    private final Type type;

    public Question(IDENTIFIER identifier, String label, Type type, CodeLocation location) {
        super(location);
        this.identifier = identifier;
        this.label = label;
        this.type = type;
    }

    public IDENTIFIER getIdentifier() {
        return identifier;
    }

    public String getLabel() {
        return label;
    }

    public Type getType() {
        return type;
    }
    /*
    public Value getDefaultValue() {
        return type.getDefaultValue();
    }
    */
    @Override
    public <T, U> T accept(AllVisitors<T, U> visitor, U context) {
        return visitor.visit(this, context);
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof Question)) {
            return false;
        }
        Question q2 = (ExpressionQuestion) obj;
        return identifier.equals(q2.getIdentifier());
    }

    @Override
    public int hashCode() {
        return identifier.hashCode();
    }

}

File: manuelalex/src/statements/Question.js
/**
 * Created by Manuel on 13/02/2017.
 */

import {Statement}      from './Statement.js';


export class Question extends Statement {
    constructor(label = {}, property = '', propertyType = null, location) {
        super(location);

        this.label = label;
        this.property = property;
        this.propertyType = propertyType;
    }

    getLabel() {
        return this.label;
    }

    getProperty() {
        return this.property;
    }

    getPropertyType() {
        return this.propertyType.getType();
    }

    accept(visitor, ...params) {
        visitor.visitQuestion(this, ...params);
    }
}